server/src/test/java/org/opensearch/common/settings/SettingTests.java

```diff
@@ -1277,6 +1363,74 @@ public void testDoubleWithMinMax() {
         assertEquals(1.2, doubleSetting.get(Settings.EMPTY).doubleValue(), 0.01);
     }
 
+    public void testDoubleParser() throws Exception {
+        String expectedKey = "test key";
+        double expectedMinValue = Double.MIN_VALUE;
+        double expectedMaxValue = Double.MAX_VALUE;
+        boolean expectedFilteredStatus = true;
+        DoubleParser doubleParser = new DoubleParser(expectedMinValue, expectedMaxValue, expectedKey, expectedFilteredStatus);
+
+        assertEquals(expectedKey, doubleParser.getKey());
+        assertEquals(expectedMinValue, doubleParser.getMin(), 0.01);
+        assertEquals(expectedMaxValue, doubleParser.getMax(), 0.01);
+        assertEquals(expectedFilteredStatus, doubleParser.getFilterStatus());
+
+        try (BytesStreamOutput out = new BytesStreamOutput()) {
+            doubleParser.writeTo(out);
+            out.flush();
+            try (BytesStreamInput in = new BytesStreamInput(BytesReference.toBytes(out.bytes()))) {
+                doubleParser = new DoubleParser(in);
+                assertEquals(expectedKey, doubleParser.getKey());
+                assertEquals(expectedMinValue, doubleParser.getMin(), 0.01);
+                assertEquals(expectedMaxValue, doubleParser.getMax(), 0.01);
+                assertEquals(expectedFilteredStatus, doubleParser.getFilterStatus());
+            }
+        }
+    }
+
+    // Boolean
+    public void testBooleanParser() throws Exception {
+        String expectedKey = "test key";
+        Boolean expectedFilteredStatus = true;
+        BooleanParser booleanParser = new BooleanParser(expectedKey, expectedFilteredStatus);
+
+        assertEquals(expectedKey, booleanParser.getKey());
+        assertEquals(expectedFilteredStatus, booleanParser.getFilterStatus());
+
+        try (BytesStreamOutput out = new BytesStreamOutput()) {
+            booleanParser.writeTo(out);
+            out.flush();
+            try (BytesStreamInput in = new BytesStreamInput(BytesReference.toBytes(out.bytes()))) {
+                booleanParser = new BooleanParser(in);
+                assertEquals(expectedKey, booleanParser.getKey());
+                assertEquals(expectedFilteredStatus, booleanParser.getFilterStatus());
+            }
+        }
+    }
+
+    // ByteSizeValue
+    public void testByteSizeValueParser() throws Exception {
```

Comment:
I don't see a test for the `MemorySizeValueParser`

Reply:
ACK